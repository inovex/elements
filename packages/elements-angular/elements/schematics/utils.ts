import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
