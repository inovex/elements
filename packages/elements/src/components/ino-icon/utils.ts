import { getAssetPath } from '@stencil/core';

let CACHED_MAP: Map<string, string>;

export function getIconMap(): Map<string, string> {
	if (!CACHED_MAP) {
		const win = window as any;
		win.inoIcons = win.inoIcons || {};
		CACHED_MAP = win.inoIcons.map = win.inoIcons.map || new Map();
	}
	return CACHED_MAP;
}

export function addIcons(icons: { [name: string]: string }) {
	const map = getIconMap();
	Object.keys(icons).forEach((name) => map.set(name, icons[name]));
}

export const getSrc = (src: string | undefined) => {
	if (isStr(src)) {
		src = src.trim();
		if (isSrc(src)) {
			return src;
		}
	}
	return null;
};

export function getName(name: string | undefined) {
	if (name && !isSrc(name)) {
		return name;
	}
	return undefined;
}

export function getUrl(src: string | undefined, icon: string | undefined) {
	let url = getSrc(src);

	if (url) {
		return url;
	}
	url = getName(icon);
	if (url) {
		return getNamedUrl(url);
	}

	url = getSrc(icon);
	if (url) {
		return url;
	}

	return null;
}

export function getNamedUrl(name: string) {
	const url = getIconMap().get(name);
	if (url) {
		return url;
	}
	return getAssetPath(`./ino-icon/${name}.svg`);
}

export const isSrc = (str: string) => str.length > 0 && /(\/|\.)/.test(str);

export const isStr = (val: any): val is string => typeof val === 'string';

export const toLower = (val: string) => val.toLowerCase();
