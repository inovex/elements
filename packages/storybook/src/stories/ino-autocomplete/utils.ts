import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
