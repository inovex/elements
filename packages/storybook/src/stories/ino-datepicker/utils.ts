import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
