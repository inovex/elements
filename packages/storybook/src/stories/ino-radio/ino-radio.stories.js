import { boolean, text } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import componentReadme from '_local-elements/src/components/ino-radio/readme.md';
import './ino-radio.scss';

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-radio') {
      return;
    }

    el.setAttribute('checked', true);
  };

  document.addEventListener('checkedChange', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('checkedChange', eventHandler);
  };
}

export default {
  title: 'Input/<ino-radio>',
  parameters: {
    actions: {
      handles: ['checkedChange ino-radio'],
    },
  },
  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
<div class="story-radio">
  <style>
          ino-radio.customizable-radio {
            --radio-unchecked-color: ${text(
              '--radio-unchecked-color',
              '#c1c1c1',
              'Custom Properties'
            )};
            --radio-checked-color: ${text(
              '--radio-checked-color',
              '#3d40f5',
              'Custom Properties'
            )};
            --radio-hover-color: ${text(
              '--radio-hover-color',
              '#5d60f7',
              'Custom Properties'
            )};
            --radio-active-color: ${text(
              '--radio-active-color',
              '#0d10f3',
              'Custom Properties'
            )};
            --radio-disabled-inner-circle-color: ${text(
              '--radio-disabled-inner-circle-color',
              '#777777',
              'Custom Properties'
            )};
            --radio-disabled-outer-circle-color: ${text(
              '--radio-disabled-outer-circle-color',
              '#c1c1c1',
              'Custom Properties'
            )};
          }
      </style>
        <ino-radio
          class="customizable-radio"
    checked="${boolean('checked', false)}"
    disabled="${boolean('disabled', false)}"
    name="radio-custom"
  >
    ${text('<slot />', 'Customizable Radio Button')}
  </ino-radio>
</div>

<h4>States</h4>
<div class="story-radio">
  <ino-radio>Unchecked</ino-radio>
  <ino-radio checked>Checked</ino-radio>
  <ino-radio disabled>Disabled</ino-radio>
  <ino-radio checked disabled>Checked and Disabled</ino-radio>
</div>
`;
