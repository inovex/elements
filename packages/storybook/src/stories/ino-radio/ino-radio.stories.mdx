import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-radio/readme.md";

<Meta title="Input/ino-radio/Documentation" />

<Description markdown={readme} />
