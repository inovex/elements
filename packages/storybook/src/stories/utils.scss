/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
