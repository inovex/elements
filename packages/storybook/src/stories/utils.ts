import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
