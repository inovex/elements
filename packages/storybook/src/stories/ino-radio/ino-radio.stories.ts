import { Components } from '@inovex.de/elements';
import { html } from 'lit-html';
import { useEffect } from '@storybook/client-api';
import { defaultDecorator } from '../utils';
import './ino-radio.scss';
import { Story } from '@storybook/web-components';

export default {
  title: 'Input/ino-radio',
  component: 'ino-radio',
  parameters: {
    actions: {
      handles: ['checkedChange ino-radio']
    }
  },
  decorators: [
    story => defaultDecorator(story, 'story-radio'), 
    story => {
      useEffect(() => {
        const eventHandler = e => e.target.setAttribute('checked', e.detail);
        const radios = document.querySelectorAll('ino-radio');
        radios.forEach(r => r.addEventListener('checkedChange', eventHandler));
        return () => radios.forEach(r => r.removeEventListener('checkedChange', eventHandler));
      });
      return story();
    }
  ],
};

export const Playground: Story<Components.InoRadio> = args => html`
  <ino-radio
    checked="${args.checked}"
    disabled="${args.disabled}"
    name="${args.name}"
    value="${args.value}"
  >
    Radio Button Label
  </ino-radio>
`;
Playground.args = {
  checked: false,
  disabled: false,
  name: 'radio-custom',
  value: 'radio-1'
};

export const States = () => html`
  <ino-radio>Unchecked</ino-radio>
  <ino-radio checked>Checked</ino-radio>
  <ino-radio disabled>Disabled</ino-radio>
  <ino-radio checked disabled>Checked and Disabled</ino-radio>
`;

// TODO: CSS Variables
//   <style>
//     ino-radio.customizable-radio {
//       --ino-radio-unchecked-color: ${text('--ino-radio-unchecked-color', '#D4DDEA', 'Custom Properties')};
//       --ino-radio-checked-color: ${text('--ino-radio-checked-color', '#EB002D', 'Custom Properties')};
//       --ino-radio-hover-color: ${text('--ino-radio-hover-color', '#FF1A47', 'Custom Properties')};
//       --ino-radio-active-color: ${text('--ino-radio-active-color', '#D20014', 'Custom Properties')};
//       --ino-radio-disabled-inner-circle-color: ${text('--ino-radio-disabled-inner-circle-color', '#777777', 'Custom Properties')};
//       --ino-radio-disabled-outer-circle-color: ${text('--ino-radio-disabled-outer-circle-color', '#c1c1c1', 'Custom Properties')};
//     }
//   </style>
