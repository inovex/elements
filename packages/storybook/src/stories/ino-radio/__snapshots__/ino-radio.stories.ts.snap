// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-radio Checked smoke-test 1`] = `
<div id="root-inner">
  <div class="story-radio">
    <ino-radio checked="true"
               disabled="false"
               name="radio-custom"
               value="radio-1"
    >
      Radio Button Label
    </ino-radio>
  </div>
</div>
`;

exports[`Input/ino-radio Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-radio">
    <ino-radio checked="false"
               disabled="true"
               name="radio-custom"
               value="radio-1"
    >
      Radio Button Label
    </ino-radio>
  </div>
</div>
`;

exports[`Input/ino-radio Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-radio">
    <ino-radio checked="false"
               disabled="false"
               name="radio-custom"
               value="radio-1"
    >
      Radio Button Label
    </ino-radio>
  </div>
</div>
`;

exports[`Input/ino-radio Unchecked smoke-test 1`] = `
<div id="root-inner">
  <div class="story-radio">
    <ino-radio checked="false"
               disabled="false"
               name="radio-custom"
               value="radio-1"
    >
      Radio Button Label
    </ino-radio>
  </div>
</div>
`;
